core/tests/webdriverio_desktop/contributorAdminDashboard.js

```diff
@@ -0,0 +1,351 @@
+    );
+    await explorationEditorPage.saveChanges('Done!');
+    await workflow.publishExploration();
+    let dummyExplorationId = await general.getExplorationIdFromEditor();
+
+    // Adding the exploration to a curated lesson.
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.waitForTopicsToLoad();
+    await topicsAndSkillsDashboardPage.navigateToTopicWithIndex(0);
+    await topicEditorPage.createStory(
+      'Story Title', 'topicandstoryeditorone', 'Story description',
+      Constants.TEST_SVG_PATH);
+    await storyEditorPage.createNewChapter(
+      'Chapter 1', dummyExplorationId, Constants.TEST_SVG_PATH);
+    await storyEditorPage.updateMetaTagContent('story meta tag');
+    await storyEditorPage.saveStory('Saving Story');
+    await storyEditorPage.publishStory();
+
+    // Testing the copy tool.
+    let opportunityActionButtonCss = $(
+      '.e2e-test-opportunity-list-item-button');
+    let copyButton = $('.e2e-test-copy-button');
+    let doneButton = $('.e2e-test-close-rich-text-component-editor');
+    let saveButton = $('.e2e-test-save-button');
+    let textbox = $('.e2e-test-description-box');
+
+    await contributorDashboardPage.get();
+    await waitFor.pageToFullyLoad();
+    await contributorDashboardPage.navigateToTranslateTextTab();
+    await contributorDashboardTranslateTextTab.changeLanguage(
+      'shqip (Albanian)');
+    await contributorDashboardPage.waitForOpportunitiesToLoad();
+    await action.click('Opportunity button', opportunityActionButtonCss);
+    let image = $('.e2e-test-image');
+    await waitFor.visibilityOf(
+      image,
+      'Test image taking too long to appear.');
+    let images = await $$('.e2e-test-image');
+    expect(images.length).toEqual(1);
+
+    // Copy tool should copy image on pressing 'Done'.
+    await waitFor.visibilityOf(
+      copyButton, 'Copy button taking too long to appear');
+    await action.click('Copy button', copyButton);
+    await action.click('Image', images[0]);
+    await action.setValue(
+      'Set Image description', textbox, 'An example description');
+    await action.click('Done', doneButton);
+    await action.click('Save', saveButton);
+    await users.logout();
+
+    // Accept suggestion as user1.
+    await users.login(USER_EMAILS[1]);
+    await contributorDashboardPage.get();
+    await contributorDashboardPage.selectTranslationReviewTab();
+    await contributorDashboardPage.waitForOpportunitiesToLoad();
+    await contributorDashboardPage.selectReviewLanguage('shqip (Albanian)');
+
+    if (width < 700) {
+      let opportunityItemSelector = function() {
+        return $$('.e2e-test-opportunity-list-item');
+      };
+      let opportunity = (await opportunityItemSelector())[0];
+      await action.click('Opportunity Item', opportunity);
+    } else {
+      await contributorDashboardPage.clickOpportunityActionButton(
+        'Chapter 1', 'Topic 0 for contribution - Story Title');
+    }
```

Comment:
WHy are we having the width check here? (Don't we have control over width when running e2e tests?)

If we want to test behavior for different screens, can't we have different describe/it blocks for each scenario?

Reply:
It's okay to use existing code but at your own responsibility to make sure you understand what's going on and only use the required steps. Can we remove things that are not required for this test setup?

Reply:
Removed it.